presto-main/src/main/java/io/prestosql/server/protocol/QueryResultRows.java

```diff
@@ -90,8 +92,13 @@ public boolean isEmpty()
         return totalRows == 0;
     }
 
+    @Nullable
     public List<Column> getColumns()
     {
+        if (columns.isEmpty()) {
```

Comment:
I do not know this part of code so I do not want to make final call. 
The question which comes to my mind is: are there legitimate cases when column list is empty and we want to distinguish it from null?

Reply:
Initial query results does not contain column list at all. Previously null was returned, now it's empty array which breaks some clients (as null is part of the protocol)